test/e2e.sh

```diff
@@ -0,0 +1,95 @@
+#! /bin/bash
+
```

Comment:
How about `set -euxo pipefail`?

Reply:
+1, except the `set -e`, or with a rewriting of the `$?` usage!

Reply:
Thanks, -ux and -o pipefail seem like good options to add.

Reply:
`-u` and '-o pipefail' are definitely good settings.
I don't think I want to add -x unless I also add -e and get rid of exit_on_error and all references to it. Otherwise, the logs will be a bit of a mess. Do you think that is what I should do? 

My goal here is to make it very easy to look at the logs of a failed test and see where it failed and why... so the verbose logging seems unhelpful unless we expect errors from the script itself, and not just failed test cases.

Reply:
> My goal here is to make it very easy to look at the logs of a failed test and see where it failed and why.

the thing is, when the test fails, you need to know if 1/ the test script failed (stupid bug) or 2/ the test is correct and OpenShift failed (real problem)

that's what I like with Ansible, you describe each step (task) with a `name` and you get the `command` being executed and the result. Most if the time it helps knowing what is faulty (the command or the command result)

so here, either you add enough `echo` logs to make it obvious what is going on, or `set -x` ...
but overall the scripts are quite simple, so on my side with fine with any of these options

Reply:
Thanks. In the interest of making sure we know what's going on in the case of stupid bugs in the script, I added -e and -x, and removed exit_on_error(). This also allowed me to remove several `echo` logs, making the whole script a bit shorter over all.

Reply:
Thank you!